slicedimage/backends/_caching.py

```diff
@@ -9,23 +12,23 @@ class CachingBackend(Backend):
     def __init__(self, cacheroot, authoritative_backend):
         self._cacheroot = cacheroot
         self._authoritative_backend = authoritative_backend
+        self.cache = Cache(cacheroot, size_limit=int(5e9))
 
     def read_file_handle_callable(self, name, checksum_sha1=None, seekable=False):
-        # TODO: (ttung) This is a very very primitive cache.  Should make this more robust with
-        # evictions and all that good stuff.
-        cachedir = os.path.join(self._cacheroot, checksum_sha1[0:2], checksum_sha1[2:4])
-        cachepath = os.path.join(cachedir, checksum_sha1)
-        if not os.path.exists(cachepath):
-            os.makedirs(cachedir)
-            with open(cachepath, "w") as dfh, \
-                    self._authoritative_backend.read_file_handle(name, checksum_sha1) as sfh:
-                while True:
-                    data = sfh.read(128 * 1024)
-                    if len(data) == 0:
-                        break
-                    dfh.write(data)
-
-        return lambda: open(cachepath, "r")
+        def returned_callable():
+            if checksum_sha1:
+                if not self.cache.get(checksum_sha1):
+                    sfh = self._authoritative_backend.read_file_handle(name)
+                    self.cache.set(checksum_sha1, sfh.data)
+                file_data = self.cache.read(checksum_sha1)
+                # The DiskCache library returns the cache data as bytes instead of a buffered reader
```

Comment:
proposal:

`If the data is small enough, the DiskCache library returns the cache data as bytes instead of a buffered reader.  In that case, we want to wrap it in a file-like object.`